### PR TITLE
PVS/V547: Expression 'eap->line2 < 0' is always false.

### PR DESCRIPTION
**Pull request by dundargoc**

### CODE REVIEW

File: src/nvim/ex_docmd.c [hunk 1]
```diff
@@ -4184,9 +4184,6 @@ static char_u *invalid_range(exarg_T *eap)
         }
         break;
       case ADDR_UNSIGNED:
-        if (eap->line2 < 0) {
```
Comment (bfredl) [inline src/nvim/ex_docmd.c:4187]: hmm what does `ADDR_UNSIGNED` even mean if all ranges are unsigned? can the flag be removed completely?

Comment (DundarGoc) [reply src/nvim/ex_docmd.c:4187]: Are you asking me specifically or just in general? I have no damn clue but I can check around to figure out what it's supposed to be. The weird thing is that I can't seem to find it's ever assigned a value. It says in ex_cmds_defs.h that ADDR_UNSIGNED defaults to 1 but I can't find it.

File: src/nvim/ex_docmd.c [hunk 2]
```diff
@@ -4184,9 +4184,6 @@ static char_u *invalid_range(exarg_T *eap)
...
-          return (char_u *)_(e_invrange);
-        }
         break;
```
Comment (janlazo) [inline src/nvim/ex_docmd.c:4187]: Delete this line if you're doing this. `ADDR_NONE` and `ADDR_UNSIGNED`  cases are grouped in other switch/case code.

File: src/nvim/ex_docmd.c (hunk 1 above)
Comment (janlazo) [reply src/nvim/ex_docmd.c:4187]: Ranges can be negative. `ADDR_UNSIGNED` prevents negative range for quickfix commands. Patch v8.1.1261, 35dc6d6e.

**Comment by janlazo** [comment]
Vim has the same error so this can be upstreamed.

File: src/nvim/ex_docmd.c (hunk 2 above)
Comment (DundarGoc) [reply src/nvim/ex_docmd.c]: I assume you mean the line break line? OK, I've removed it. Thanks for the help.

**Comment by DundarGoc** [comment]
@janlazo @bfredl I pressed the rotate function to see what it did before I read it, I didn't know it would send a review request :) . You can ignore it.

**Comment by janlazo** [comment]
@dundargoc Bram ported your changes so create a new vim-patch PR or redo your commit here as a vim-patch commit.

**Comment by jimman2003** [comment]
Vim-Patch: #14866

**Pull request closed without merging**